crates/solver/src/settlement_rater.rs

```diff
@@ -30,19 +30,44 @@ use {
     web3::types::AccessList,
 };
 
-struct SimulationSuccess {
-    pub simulation: Simulation,
-    pub gas_estimate: U256,
+type GasEstimate = U256;
+
+pub enum SimulateError {
+    FailedSimulation(SimulationWithError),
+    Internal(anyhow::Error),
+}
+
+impl From<anyhow::Error> for SimulateError {
```

Comment:
How do others feel about these "blanket" castings from anyhow::Error? In my experience it makes it very easy for people to not think about how their error should be handled (putting a ? on it will make it bubble up), whereas if they have to explicitly decide that this is an InternalErrror (ie. not some other type that should be handled in a certain way) we have less room for "accidental error handling"

Reply:
I agree that having explicit errors forces you to put more thought into each error case.
However, I find reviewing the code that converts from one error type to another quite tricky to review and follow in the code. This has mostly been an issue for me in the orderbook and less so in the new driver which also does it so maybe there is a good way to do it.
However, I would keep that change for a separate PR because that can touch a bunch of code blowing up the scope of this PR.

Reply:
We used `anyhow::Error` to indicate opaque errors. I think if there is an error worth special handling, it shouldn't ever be represented as an `anyhow::Error` because of type erasure.

Reply:
I also tend to agree that `anyhow::Error` seems super convenient, and it really is. But as the application starts becoming more complex, it ends up doing more harm than good. I like that the `driver` doesn't use `anyhow::Error` at all (even if it forces more boilerplate).

`thiserror` on the other hand is a gift from the heavens.